pkg/plugins/placementpolicy/core/core.go

```diff
@@ -73,38 +57,114 @@ func (m *PlacementPolicyManager) GetPlacementPolicyForPod(ctx context.Context, p
 	return ppList[0], nil
 }
 
-func (m *PlacementPolicyManager) GetPodsWithLabels(ctx context.Context, podLabels map[string]string) ([]*corev1.Pod, error) {
-	return m.podLister.List(labels.Set(podLabels).AsSelector())
+func (m *PlacementPolicyManager) filterPlacementPolicyList(ppList []*v1alpha1.PlacementPolicy, pod *corev1.Pod) []*v1alpha1.PlacementPolicy {
+	var filteredPPList []*v1alpha1.PlacementPolicy
+	for _, pp := range ppList {
+		labels := pp.Spec.PodSelector.MatchLabels
+		if utils.HasMatchingLabels(pod.Labels, labels) {
+			filteredPPList = append(filteredPPList, pp)
+		}
+	}
+	return filteredPPList
 }
 
-// AnnotatePod annotates the pod with the placement policy.
-func (m *PlacementPolicyManager) AnnotatePod(ctx context.Context, pod *corev1.Pod, pp *v1alpha1.PlacementPolicy, preferredNodeWithMatchingLabels bool) (*corev1.Pod, error) {
-	annotations := map[string]string{}
-	if pod.Annotations != nil {
-		annotations = pod.Annotations
+func (m *PlacementPolicyManager) GetPolicyInfo(pp *v1alpha1.PlacementPolicy) *PolicyInfo {
+	policyNamespace := pp.Namespace
+	policyName := pp.Name
+	corePolicy := pp.Spec.Policy
+
+	namespacePolicies, namespaceExists := m.policies[policyNamespace]
```

Comment:
I think it'll be good to have a `GET` method on `PolicyInfos` which uses a mutex internally to access the map and return the `PolicyInfo`. This will prevent any race conditions caused by concurrent read/write.

Reply:
👍